x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/search/SparseVectorQueryBuilder.java

```diff
@@ -222,28 +226,68 @@ protected Query doToQuery(SearchExecutionContext context) throws IOException {
             );
         }
 
-        return (shouldPruneTokens)
-            ? WeightedTokensUtils.queryBuilderWithPrunedTokens(fieldName, tokenPruningConfig, queryVectors, ft, context)
+        // if the query options for pruning are not set,
+        // we need to check the index options for this field
+        // and use those if set.
+        boolean doPruneTokens = false;
+        TokenPruningConfig setTokenPruningConfig = tokenPruningConfig;
+
+        if (shouldPruneTokens == null || setTokenPruningConfig == null) {
+            SparseVectorFieldMapper sparseVectorFieldMapper = getSparseVectorFieldMapperForSearchExecution(fieldName, context);
+            IndexFieldPruningSettings indexPruningSettings = getIndexFieldPruningSettings(sparseVectorFieldMapper);
+            if (shouldPruneTokens == null && indexPruningSettings.prune != null && indexPruningSettings.prune) {
+                doPruneTokens = true;
+            }
+            if (setTokenPruningConfig == null && indexPruningSettings.pruningConfig != null) {
+                setTokenPruningConfig = indexPruningSettings.pruningConfig;
+            }
+        } else {
+            doPruneTokens = shouldPruneTokens;
+        }
+
+        return (doPruneTokens)
+            ? WeightedTokensUtils.queryBuilderWithPrunedTokens(fieldName, setTokenPruningConfig, queryVectors, ft, context)
             : WeightedTokensUtils.queryBuilderWithAllTokens(fieldName, queryVectors, ft, context);
     }
 
     @Override
     protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) {
         if (queryVectors != null) {
             return this;
-        } else if (weightedTokensSupplier != null) {
+        }
+
+        if (weightedTokensSupplier != null) {
             TextExpansionResults textExpansionResults = weightedTokensSupplier.get();
             if (textExpansionResults == null) {
                 return this; // No results yet
             }
 
+            // if the query options for pruning are not set,
+            // we need to check the index options for this field
+            // and use those if set.
+            boolean doPruneTokens = false;
+            TokenPruningConfig setTokenPruningConfig = tokenPruningConfig;
+
+            if (shouldPruneTokens == null || setTokenPruningConfig == null) {
```

Comment:
Looks like there is some duplicate logic here that could potentially be broken out.